modules/web/server/src/main/scala/observe/web/server/http4s/StaticRoutes.scala

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package observe.web.server.http4s
+
+import scala.concurrent.duration._
+
+import cats.data.NonEmptyList
+import cats.data.OptionT
+import cats.effect.Sync
+import cats.syntax.all.*
+import org.http4s.CacheDirective._
+import org.http4s.HttpRoutes
+import org.http4s.Request
+import org.http4s.Response
+import org.http4s.StaticFile
+import org.http4s.headers.`Cache-Control`
+import org.http4s.server.middleware.GZip
+import org.http4s.Header
+import org.http4s.Uri
+import fs2.compression.Compression
+import fs2.io.file.Files
+import java.nio.file.Paths
+import fs2.io.file.Path
+
+class StaticRoutes[F[_]: Sync: Compression: Files]:
+  private val AppDir: String = "app"
+
+  private val OneYear: Int = 365 * 24 * 60 * 60 // One year in seconds
+
+  private val CacheHeaders: List[Header.ToRaw] = List(
+    `Cache-Control`(NonEmptyList.of(`max-age`(OneYear.seconds)))
+  )
+
+  def localFile(path: String, req: Request[F]): OptionT[F, Response[F]] =
+    // Get full path to JAR/class
+    // https://stackoverflow.com/questions/320542/how-to-get-the-path-of-a-running-jar-file
+    val runningPath = Paths.get(getClass.getProtectionDomain.getCodeSource.getLocation.toURI)
+    // Get the current directory (should be "lib") and then go to the "app" sibling
+    val fullPath    = runningPath.getParent.resolveSibling(AppDir).resolve(path.stripPrefix("/"))
+    StaticFile.fromPath(Path.fromNioPath(fullPath), req.some) // .map(_.putHeaders())
+
+  implicit class ReqOps(req: Request[F]) {
```

Comment:
No extension?

Reply:
WDYM?

Reply:
Could this be an extension method?

Reply:
Ah, yes! This was copied from seqexec. I'll migrate.